Update getting-started URLs within reference documentation

jayeff opened the pull request:
## Summary of the changes / Why this improves CrateDB

The getting-started urls are all redirecting to https://crate.io/docs/crate/reference/en/4.5/index.html but should link to the tutorial section.

IMO within the docs we should use the correct URLs. I will create a separate issue to look into the redirect URLs from old `getting-started` to new `tutorials section.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

jayeff commented on docs/config/environment.rst:
```diff
@@ -110,7 +110,7 @@ General
   - For `basic installations`_, the process working directory
 
-  - If you have installed `a CrateDB Linux package`_, ``/var/lib/crate``
+  - If you have installed a CrateDB Linux package, ``/var/lib/crate``
```
The new installation tutorial has two sections for CrateDB Linux package installation:
* https://crate.io/docs/crate/tutorials/en/latest/install.html#debian-and-ubuntu
* https://crate.io/docs/crate/tutorials/en/latest/install.html#red-hat-linux-and-centos

This makes it a bit complicated to link to general Linux package. I removed the link for now as I wasn't sure what would be best here.

Any suggestion where to link to?

amotl replied:
Yeah. While we now have everything on a single page, canonically linking to "Linux" is currently not possible. It's a balancing act.

For me, it is either fine to keep it like it is now or otherwise improve this detail again. Maybe @proddata and @mfussenegger want to add some thoughts here.

mfussenegger replied:
We could create a heading hierarchy. 

```
# Install via Linux package

## Debian or Ubuntu

## Redhat or Centos
```

amotl replied:
Sure, I can do that if linking there would be more convenient.

amotl replied:
Implemented by https://github.com/crate/crate-tutorials/pull/53.

amotl replied:
You can now link to the Linux section by using https://crate.io/docs/crate/tutorials/en/latest/install.html#linux.

jayeff commented on docs/config/environment.rst:
```diff
@@ -147,11 +147,9 @@ logging of the JVM.
-.. _expanded tarball: https://crate.io/docs/crate/getting-started/en/latest/install-run/basic.html
+.. _expanded tarball: https://crate.io/docs/crate/tutorials/en/latest/install.html#ad-hoc-installation-unix-windows
 .. _Concurrent Mark Sweep: https://docs.oracle.com/javase/10/gctuning/concurrent-mark-sweep-cms-collector.htm
 .. _G1: https://docs.oracle.com/javase/10/gctuning/garbage-first-garbage-collector.htm
```
Should we update these links to a newer Java SE version?

amotl replied:
Sure, why not.

- https://docs.oracle.com/en/java/javase/11/gctuning/concurrent-mark-sweep-cms-collector.html
- https://docs.oracle.com/en/java/javase/16/gctuning/garbage-first-g1-garbage-collector1.html

> The CMS collector is deprecated. Strongly consider using the Garbage-First collector instead.

jayeff replied:
@amotl I updated the URLs and noted that CMS is deprecated

jayeff commented on docs/config/environment.rst:
```diff
@@ -147,11 +147,9 @@ logging of the JVM.
-.. _a CrateDB Linux package: https://crate.io/docs/crate/getting-started/en/latest/install-run/special/linux.html
-.. _CrateDB on Docker: https://crate.io/docs/crate/getting-started/en/latest/install-run/special/docker.html
-.. _Java options: https://docs.oracle.com/javase/7/docs/technotes/tools/windows/java.html#CBBIJCHG
+.. _basic installations: https://crate.io/docs/crate/tutorials/en/latest/install.html
```
Do we still have a "basic installation"? I think this is different now as to how it was explained in earlier versions of the installation guide.

amotl replied:
> Do we still have a "basic installation"?

The thing which would fit best has been rephrased to "ad-hoc installation", see https://crate.io/docs/crate/tutorials/en/latest/install.html#ad-hoc-installation-unix-windows.

Whether we should bring back the other original variant outlining the usage of https://try.crate.io/ will be discussed at https://github.com/crate/crate-tutorials/issues/51.

amotl replied:
The new link for reflecting the "basic" aka. "tarball" installation flavor would be https://crate.io/docs/crate/tutorials/en/latest/install.html#install-adhoc.

jayeff commented:
@amotl Wow. this review was fast 🚀 

I added a couple questions. Can you check them out?

mfussenegger commented on docs/config/environment.rst:
```diff
@@ -128,9 +128,10 @@ Collector
-Before CrateDB 4.1 it defaulted to use the `Concurrent Mark Sweep` garbage
-collector. If you'd like to continue using CMS, you can switch setting the
-following :ref:`CRATE_JAVA_OPTS <conf-env-java-opts>`::
+Before CrateDB 4.1 it defaulted to use the `Concurrent Mark Sweep`_ garbage
+collector. Please note that CMS garbage collection is deprecated. If you'd
+like to continue using CMS, you can switch the following
+:ref:`CRATE_JAVA_OPTS <conf-env-java-opts>`: setting:
```
I think we've to update this. CMS has been removed with JDK 14 https://openjdk.java.net/jeps/363

amotl reviewed (changes requested):
Hi Johannes,

I've adjusted `crate-tutorials` so that it now has appropriate link anchors for your convenience.

With kind regards,
Andreas.

jayeff commented:
@amotl I updated the PR and rebased on current master.

I choose to remove the change regarding CMS GC here as it's not related to getting-started URLs
